owncloudApp/src/main/java/com/owncloud/android/operations/UploadFileOperation.java

```diff
@@ -477,16 +477,16 @@ protected void moveTemporalOriginalFiles(File temporalFile, File originalFile, S
      * @return      'True' if the upload was delayed until WiFi connectivity is available, 'false' otherwise.
      */
     private boolean delayForWifi() {
-        boolean delayCameraUploadsPicture = (
-                isCameraUploadsPicture() &&  PreferenceManager.cameraPictureUploadViaWiFiOnly(mContext)
-        );
-        boolean delayCameraUploadsVideo = (
-                isCameraUploadsVideo() && PreferenceManager.cameraVideoUploadViaWiFiOnly(mContext)
-        );
-        return (
-                (delayCameraUploadsPicture || delayCameraUploadsVideo) &&
-                        !ConnectivityUtils.isAppConnectedViaWiFi(mContext)
-        );
+        switch(getCreatedBy()){
+            case CREATED_AS_CAMERA_UPLOAD_PICTURE:
+                return PreferenceManager.cameraPictureUploadViaWiFiOnly(mContext);
+            case CREATED_AS_CAMERA_UPLOAD_VIDEO:
+                return PreferenceManager.cameraVideoUploadViaWiFiOnly(mContext);
+            case CREATED_BY_USER:
+                return !ConnectivityUtils.isAppConnectedViaWiFi(mContext);
+        }
+        //cannot happen
```

Comment:
if it can not happen (the three options above are the only options) , it is not needed to add the `return false`.
i'd also add a `break` inside each `case` to avoid checking the other options, once one of the is matched. What do you think? CC @davigonz

Reply:
> if it can not happen (the three options above are the only options) , it is not needed to add the return false.

You are right, I missed that part of the code, I guess @BigAdam2005 put that return there because delayForWifi has to return a boolean even if any of the cases do not match the switch. To fix this, please add a `default` case at the end of the switch. @BigAdam2005 

`break` would not be needed since the method finish when returning in any of the cases.

Reply:
It isn`t possible because i got compilation error
`error: missing return statement`

Reply:
```
private boolean delayForWifi() {
        switch(getCreatedBy()){
            case CREATED_AS_CAMERA_UPLOAD_PICTURE:
                return PreferenceManager.cameraPictureUploadViaWiFiOnly(mContext);
            case CREATED_AS_CAMERA_UPLOAD_VIDEO:
                return PreferenceManager.cameraVideoUploadViaWiFiOnly(mContext);
            case CREATED_BY_USER:
                return !ConnectivityUtils.isAppConnectedViaWiFi(mContext);
            default:
                return false;
        }
    }
```

Have you tried to include the `default` case below the other conditions?

Reply:
You shouldn't have `error: missing return statement.` then.